src/popup/popup.js

```diff
@@ -0,0 +1,186 @@
+ * @returns {Object} moduleList 
+ */
+async function getModuleList() {
+    let moduleList = {};
+
+    // if you reset the storage
+    const localStorage = await getLocalStorage();
+    if (Object.keys(localStorage).length === 0) {
+        const moduleList = {}
+        await setModuleList(moduleList)
+    }
+
+    if (Helpers.isFirefox()) {
+        moduleList = await browser.storage.local.get("moduleList");
+    }
+    else {
+        moduleList = await chrome.storage.local.get("moduleList");
+    }
+    return moduleList.moduleList;
+}
+
+/**
+ * 
+ * @param {Object} moduleList 
+ */
+async function setModuleList(moduleList) {
+    if (Helpers.isFirefox()) {
+        await browser.storage.local.set({ moduleList: moduleList })
+    }
+    else {
+        await chrome.storage.local.set({ moduleList: moduleList })
+    }
+}
+
+/**
+ * deletes the selected module from local storage
+ */
+async function removeCustomModule() {
+    const moduleList = await getModuleList();
+    const selectModuleList = document.getElementById("moduleList");
+
+    const selectedIndex = selectModuleList.selectedIndex
+    if (selectedIndex > -1) {
+        const selectedModule = selectModuleList.options[selectedIndex].value;
+        delete moduleList[selectedModule]
+        await setModuleList(moduleList)
+    }
+    else {
+        console.warn("select a module")
+    }
+}
+
+/**
+ * adds custom module to local storage
+ */
+async function addCustomModule() {
+    const modulAcronym = document.getElementById("modulAcronym").value;
+    const moduleType = document.getElementById("moduleType").value;
+    const modulCredits = document.getElementById("modulCredits").value;
+    const moduleGrade = document.getElementById("moduleGrade").value;
+    if (moduleGrade == "-") {
+        moduleGrade = 'n/a';
+    }
+
+    let modulYearList = document.getElementById("moduleYear")
+    let moduleYear = modulYearList.options[modulYearList.selectedIndex].value;
+
+    const moduleSemesterRadios = document.getElementsByName('moduleImplementation');
+    for (let i = 0, length = moduleSemesterRadios.length; i < length; i++) {
+        if (moduleSemesterRadios[i].checked) {
+            moduleSemester = moduleSemesterRadios[i].value;
+            break;
+        }
+    }
+
+    let moduleMark = document.getElementById("moduleMark").value;
+    if (moduleMark < 1) {
+        moduleMark = 'n/a';
+    }
+
+    moduleList = await getModuleList();
+
+    moduleList[modulAcronym] = {
+        acronym: modulAcronym,
+        type: moduleType,
+        credits: modulCredits,
+        mark: moduleMark,
+        grade: moduleGrade,
+        year: moduleYear,
+        semster: moduleSemester
+    }
+    await setModuleList(moduleList)
+}
+
+
+window.addEventListener('DOMContentLoaded', () => {
+    document.getElementById("submitModul").addEventListener('click', addCustomModule);
+    document.getElementById("removeModule").addEventListener('click', removeCustomModule);
```

Comment:
you can use `window.onclick = () => ...` here: https://www.w3schools.com/jsref/event_onclick.asp